File: Music.Short.Android/app/src/main/java/com/zitherharpmusic/zhmshort/ui/artist/ArtistListDialog.java
package com.zitherharpmusic.zhmshort.ui.artist;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.zitherharpmusic.zhmshort.R;
import com.zitherharpmusic.zhmshort.model.Artist;

import java.util.List;

public class ArtistListDialog extends BottomSheetDialogFragment {
    private static List<Artist> artists;

    public static ArtistListDialog newInstance(List<Artist> artists) {
        ArtistListDialog.artists = artists;
        return new ArtistListDialog();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_artist_list, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        RecyclerView recyclerView = view.findViewById(R.id.recycler_view);
        recyclerView.setAdapter(new ArtistListAdapter(getActivity(), artists));
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
    }
}

File: Music.Client.Android/Kotlin/Music/short/src/main/java/com/zitherharp/music/shorts/ui/artist/ArtistListDialog.kt
package com.zitherharp.music.shorts.ui.artist

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.zitherharp.music.model.Artist
import com.zitherharp.music.R
import com.zitherharp.music.shorts.databinding.ArtistListFragmentBinding

class ArtistListDialog: BottomSheetDialogFragment {
    private lateinit var binding: ArtistListFragmentBinding
    private var artists: List<Artist> = ArrayList()

    constructor(): super()

    constructor(artists: List<Artist>): super() {
        this.artists = artists
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) =
        ArtistListFragmentBinding.inflate(inflater, container, false).apply {
            binding = this
        }.root

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        with(binding) {
            if (artists.isNotEmpty()) {
                artistList.adapter = ArtistListAdapter(view.context, artists)
            } else {
                status.text = getString(R.string.empty)
            }
        }
    }
}